js/dragDrop.js

```diff
@@ -0,0 +1,40 @@
+$(document).ready(function () {
+    dragDropOps.dragOps();
+    dragDropOps.dropOps();
+    dragDropOps.binOps();
+});
+
+var dragDropOps = {
+    dragOps: function () {
+        $(".product").draggable({
+            helper: "clone",
+            revert: "invalid"
+        });
+    },
+
+    dropOps: function() {
+        $(".cart").droppable({
+            activeClass: "ui-state-highlight",
+            hoverClass: "ui-state-hover",
+            accept: '.product:not(".clones")',
+            drop: function (event, ui) {
+                $(this).append($(ui.draggable).clone().addClass('clones'));
+                $('.clones').draggable({
```

Comment:
In the first drag operation a clone of the original has been created when it is dropped to a droppable area the original element is getting removed. But, in the next one a different approach is followed as we are not making any clones and simply dragging the original one from draggable area to drppable area. Hence a different approach is followed just for experiment.